test/controllers/ht_registrations_controller_test.rb

```diff
@@ -230,6 +230,16 @@ def setup
     assert_equal 0, HTRegistration.count
     assert_not_empty flash[:alert]
   end
+
+  test "refuses to create registration for someone already in ht_users" do
+    user = FactoryBot.create(:ht_user)
+    params = FactoryBot.build(:ht_registration, applicant_email: user.email)
```

Comment:
If memory serves I think you can use `FactoryBot.attributes_for` for this situation

Reply:
That is correct. And more concise, as it turns out. Change made.